modules/currency-l1/src/main/scala/org/tessellation/currency/l1/http/p2p/P2PClient.scala

```diff
@@ -16,7 +17,8 @@ object P2PClient {
 
   def make[F[_]: Async: SecurityProvider: KryoSerializer](
     dagL1P2PClient: DagL1P2PClient[F],
-    client: Client[F]
+    client: Client[F],
+    l0Peer: L0Peer
```

Comment:
global l0 peer from CLI should be only used to discover the network and other cluster peers once at the very start of the node. IMO it should not be a single source of truth later on. also, it's in opposite to the ticket PROT-254 (**Design process for gL0 network upgrades that prevents all metagraphs from being taken offline.**). we've already discussed on slack with @ryle-goehausen that there should be discovery process working on a CL0 and CL1 so it's possible to restart the CL0 network without restarting the CL1 (and discovery process would be able to fetch new peers based on majority selection @sdavidp is working on).

cc: @TheMMaciek @kpudlik

Reply:
The desire is to only have the trust model run on L0. The other services will query L0 for the trust information as they need it. Can you suggest a better way to achieve this, @marcinwadon @TheMMaciek @kpudlik .

Reply:
The reasoning behind this is we want to support a path to guide users to configure their initial GL0 peer as the sidecar node (i.e. the same one running locally,) so that we can distinguish this as being a 'single model source' -- instead of running the model independently on every process. The user would have to ensure the configuration for that node so that it's one they setup (which would put it on them as the config source,) It would not function for users who are NOT running a GL0 (for that, they can default back to the normal seeds if they're not supplying information or otherwise running the model.) 

It shouldn't actually affect the other discovery processes which can proceed with other peers after initial configuration.

Reply:
Per our offline discussion, @marcinwadon , a random `L0Peer` from the layer's `l0Cluster` is used as the `TrustClient`'s `L0Peer`.

Reply:
It's still vulnerable. You should pass here not a single peer but a storage for global l0 peers and select the peer when fetching

Reply:
Moved the peer selection logic into `RandomPeerSelect` which is passed all the `L0Peer`s.